Specs/Scene/ExpressionSpec.js

```diff
@@ -783,6 +922,56 @@ defineSuite([
 
         expression = new Expression('color(\'green\') != color(\'green\')');
         expect(expression.evaluate(frameState, undefined)).toEqual(false);
+
+        expression = new Expression('color(\'green\') !== color(\'green\')');
+        expect(expression.evaluate(frameState, undefined)).toEqual(false);
+    });
+
+    it('evaluates vector operations', function() {
+        var expression = new Expression('+vec3(1, 2, 3)');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(1, 2, 3, 0));
+
+        expression = new Expression('-vec3(1, 2, 3)');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(-1, -2, -3, 0));
+
+        expression = new Expression('vec2(1, 2) + vec4(3, 4, 5, 6)');
```

Comment:
> What about dropping vec2 and vec3 and always using vec4?

We could see how much milage we get from this.

Remind me, why are we adding vector types at all right now?  To flush out the math library with things like dot and cross products?  If that is the case, don't we really need a `vec3`.

I am also still not convinced that we can't reasonably detect this even if some error messages have to be "Could not compile generated GLSL shader, this is often due to a vector type mismatch."